src/client/mcp-client.ts

```diff
@@ -187,6 +191,38 @@ export class MCPClient implements IMCPClient {
         }
     }
 
+    /**
+     * Connect to an MCP server via Streamable HTTP transport
+     */
+    private async connectViaHttp(
+        baseUrl: string,
+        headers: Record<string, string>,
+        timeout: number | undefined,
```

Comment:
timeout is unused, we can remove this argument

currently `this.timeout` is used for request timeouts

we could actually have a separate connection timeout for initializing the server connection, we can do in a separate PR